solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -447,21 +450,42 @@ public Cancellable asyncRequest(
               @Override
               public void onHeaders(Response response) {
                 super.onHeaders(response);
+                if (log.isDebugEnabled()) {
+                  log.debug("response processing started");
+                }
                 InputStreamResponseListener listener = this;
+                MDCCopyHelper mdcCopyHelper = new MDCCopyHelper();
                 executor.execute(
                     () -> {
                       InputStream is = listener.getInputStream();
                       assert ObjectReleaseTracker.track(is);
                       try {
                         NamedList<Object> body =
                             processErrorsAndResponse(solrRequest, parser, response, is);
+                        mdcCopyHelper.onBegin(null);
+                        if (log.isDebugEnabled()) {
```

Comment:
Can you please remove all the needless "log.isDebugEnabled" checks?  It's needlessly verbose.  I suspect you are doing this because you are seeing it in many places but we only do this when there are interpolated arguments `{}` with non-simply references to objects.

Reply:
I was actually following the 'local style' of the class
```
      if (log.isDebugEnabled()) {
        log.debug("Create Http2SolrClient with HTTP/1.1 transport");
      }
```
but sure, I will remove shortly